src/screen/splash.rs

```diff
@@ -55,11 +66,48 @@ fn spawn_splash(mut commands: Commands, asset_server: Res<AssetServer>) {
                     )),
                     ..default()
                 },
+                UiImageFadeInOut {
+                    total_duration: SPLASH_DURATION_SECS,
+                    fade_duration: SPLASH_FADE_DURATION_SECS,
+                    t: 0.0,
+                },
             ));
         });
 }
 
-const SPLASH_DURATION_SECS: f32 = 1.2;
+#[derive(Component, Reflect)]
+#[reflect(Component)]
+struct UiImageFadeInOut {
+    /// Total duration in seconds.
+    total_duration: f32,
+    /// Fade duration in seconds.
+    fade_duration: f32,
+    /// Current progress in seconds, between 0 and [`Self::total_duration`].
+    t: f32,
+}
+
+impl UiImageFadeInOut {
+    fn alpha(&self) -> f32 {
+        // Normalize by duration.
+        let t = (self.t / self.total_duration).clamp(0.0, 1.0);
+        let fade = self.fade_duration / self.total_duration;
+
+        // Regular trapezoid-shaped graph, flat at the top with alpha = 1.0.
+        ((1.0 - (2.0 * t - 1.0).abs()) / fade).min(1.0)
+    }
+}
+
+fn tick_fade_in_out(time: Res<Time>, mut animation_query: Query<&mut UiImageFadeInOut>) {
+    for mut anim in &mut animation_query {
+        anim.t += time.delta_seconds();
```

Comment:
When working with curves you usually talk about `t` in `0..=1` range, I think the scaling should happen here

Reply:
Yeah I had it like that at one point. Scaling later makes it more ergonomic to add like an `anim.rate` field that you can adjust while the animation plays. Ofc we're not doing that, but I had to choose between scale early or scale late, so that was why I chose scale late.